src/views/Composite/CompositeViewData.ts

```diff
@@ -6,15 +6,17 @@ import { isString } from "vega"
 type CVViewData = {
     label: string
     type: string
-    figureDataSha1: Sha1Hash
+    figureDataSha1?: Sha1Hash // old
+    figureDataUri?: string // new
     defaultHeight?: number
 }
 
 const isCVViewData = (x: any): x is CVViewData => {
     return validateObject(x, {
         label: isString,
         type: isString,
-        figureDataSha1: isSha1Hash,
+        figureDataSha1: optional(isSha1Hash), // old
+        figureDataUri: optional(isString), // new
```

Comment:
Would it be worthwhile to require that one of these two be set? (Not sure if the framework allows that, and it is checked for elsewhere in the code anyway, but it seems like we have to do that check in several places otherwise)

Reply:
I'm thinking of this as a transition state... eventually remove figureDataSha1 since that should really be provided as a uri starting with sha1://...

Once all the visualization plugins have been updated to do that, we can remove the old field.